fendermint/vm/interpreter/src/fvm/state/priority.rs

```diff
@@ -0,0 +1,204 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::fvm::FvmMessage;
+use fvm_shared::bigint::BigInt;
+use fvm_shared::econ::TokenAmount;
+use num_traits::{ToPrimitive, Zero};
+
+/// The transaction priority calculator. The priority calculated is used to determine the ordering
+/// in the mempool.
+#[derive(Clone, Debug)]
+pub struct TxnPriorityCalculator {
+    /// Ring buffer of base fee history
+    base_fee_history: Vec<Option<TokenAmount>>,
```

Comment:
I think this was quite early discussion on one of the huddles to use the lowest base fee in recent history to let slightly more transaction to go through to avoid empty blocks. An array is also tuneable in how base fee can be selected. The requirement is updated slightly and base fee selection is now includable/non-includable. Then the most recent base fee should be enough. I can update this accordingly.

Reply:
We'll need to fix this. The base fee determines inclusion and the burn rate (network fee). By no means should a transaction be included if its gas fee cap is below the current base fee for the block its included in. That's a consensus fault.